api4/job.go

```diff
@@ -38,6 +42,56 @@ func getJob(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(job.ToJson()))
 }
 
+func downloadJob(c *Context, w http.ResponseWriter, r *http.Request) {
+	const FILE_PATH = "export/%s/%s"
+	fileInfo := map[string]map[string]string{
+		"csv": {
+			"fileName": "csv_export.zip",
+			"fileMime": "application/zip",
+		},
+		"actiance": {
+			"fileName": "actiance_export.xml",
+			"fileMime": "application/xml",
+		},
+	}
+
+	c.RequireJobId()
+	if c.Err != nil {
+		return
+	}
+
+	if !c.App.SessionHasPermissionTo(*c.App.Session(), model.PERMISSION_MANAGE_JOBS) {
+		c.SetPermissionError(model.PERMISSION_MANAGE_JOBS)
+		return
+	}
+
+	job, err := c.App.GetJob(c.Params.JobId)
+	if err != nil {
+		mlog.Error(err.Error())
+		c.Err = err
+		return
+	}
+
+	filePath := fmt.Sprintf(FILE_PATH, job.Id, fileInfo[job.Data["export_type"]]["fileName"])
+	fileReader, err := c.App.FileReader(filePath)
+	if err != nil {
+		mlog.Error(err.Error())
+		c.Err = err
+		c.Err.StatusCode = http.StatusNotFound
+		return
+	}
+	defer fileReader.Close()
+
+	// We are able to pass 0 for content size due to the fact that Golang's serveContent (https://golang.org/src/net/http/fs.go)
+	// already sets that for us (Line 291)
```

Comment:
Is this comment needed?

Reply:
I would keep the comment, but remove line number as depending on version, that could change.